boxes/__init__.py

```diff
@@ -1278,6 +1301,39 @@ def flatHole(self, x, y, r=None, d=None, w=None, rel_w=0.75, angle=0):
             self.edge(2*r*math.sin(math.radians(a)))
             self.corner(-a)
 
+    @restore
+    @holeCol
+    def mountingHole(self, x, y, d_shaft=0.0, d_head=0.0, angle=0, tabs=0):
+        """
+        Draw a pear shaped mounting hole for sliding over a screw head. Total height = 1.5* d_shaft + d_head
+
+        :param x: position
+        :param y: postion
+        :param d_shaft: diameter of the screw shaft
+        :param d_head: diameter of the screw head
+        :param angle: rotation angle of the hole
+
+        """
+        
+        if d_shaft < 2*self.burn:
```

Comment:
May be the check for the  shaft being zero should go here instead of being duplicated in every generator.